### PR TITLE
Ndev 2317 tests for multiple token feature

### PR DESCRIPTION
**Pull request by kristinaNikolaevaa**

### CODE REVIEW

File: integration/tests/basic/evm/opcodes/test_chainid.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class TestChainId(BaseMixin):
+    @pytest.fixture(scope="class")
+    def contract(self, web3_client, class_account):
+        contract, _ = web3_client.deploy_and_get_contract(
+            "opcodes/ChainId", "0.8.10", class_account)
+        return contract
+
+    def test_chainId(self, contract):
+        assert contract.functions.getCurrentValues().call() == 0
```
Comment (gigimon) [inline integration/tests/basic/evm/opcodes/test_chainid.py]: 0 is a right value for any network?

File: integration/tests/basic/helpers/assert_message.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 class ErrorMessage(Enum):
     NEGATIVE_VALUE = "Resulting wei value must be between 1 and "
     INSUFFICIENT_FUNDS = "insufficient funds for transfer"
```
Comment (gigimon) [inline integration/tests/basic/helpers/assert_message.py:6]: this too errors exist? do you difference?

File: integration/tests/conftest.py [hunk 3]
```diff
@@ -205,13 +205,28 @@ def erc20_spl_mintable(web3_client: NeonChainWeb3Client, faucet, sol_client, sol
...
+                                                       pytestconfig.environment.evm_loader)
+    yield account
 
 
 @pytest.fixture(scope="class")
-def class_account(web3_client, faucet, eth_bank_account):
-    yield web3_client.create_account_with_balance(faucet, bank_account=eth_bank_account)
+def class_account(web3_client, faucet, eth_bank_account, solana_account, sol_client, web3_client_sol, pytestconfig):
+    account = web3_client.create_account_with_balance(faucet, bank_account=eth_bank_account)
+    sol_client.request_airdrop(solana_account.public_key, 5 * LAMPORT_PER_SOL)
```
Comment (gigimon) [inline integration/tests/conftest.py]: this airdrop can't work in devnet with bank account. For 4 threads we don't have 20 sols in devnet

File: utils/solana_client.py [hunk 4]
```diff
@@ -65,17 +70,16 @@ def send_sol(self, from_: Keypair, to: PublicKey, amount_lamports: int):
         else:
             raise AssertionError(f"Balance not changed in account {to}")
 
-    def get_neon_account_address(
-        self, neon_account_address: str, evm_loader_id: str
-    ) -> PublicKey:
-        neon_account_addressbytes = bytes.fromhex(neon_account_address[2:])
+    def ether2balance(self, address: tp.Union[str, bytes], chain_id: int, evm_loader_id: str) -> PublicKey:
```
Comment (gigimon) [inline utils/solana_client.py:63]: add pls help here, what it return

File: integration/tests/basic/test_payment_in_different_tokens.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    def bob(self, class_account):
+        return class_account
+
+    @pytest.fixture(scope="class")
+    def alice(self, web3_client, web3_client_sol, faucet, eth_bank_account):
+        return web3_client_sol.create_account()
+
+    @pytest.fixture(scope="function")
+    def check_neon_balance_does_not_changed(self, alice, bob, web3_client):
+        alica_balance_alica_before = web3_client.get_balance(alice)
```
Comment (gigimon) [inline integration/tests/basic/test_payment_in_different_tokens.py]: alica_balance_alica_before -> alicE_balance_before

File: integration/tests/basic/test_payment_in_different_tokens.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
+import allure
+import pytest
+
+from integration.tests.basic.helpers.basic import BaseMixin
+
+
+@allure.feature("Multiply token")
+@allure.story("Payments in sol tokens")
+class TestSolChain(BaseMixin):
```
Comment (gigimon) [inline integration/tests/basic/test_payment_in_different_tokens.py:12]: we should ru nthis tests only if this network has "sol" network

File: integration/tests/basic/test_payment_in_different_tokens.py [hunk 7]
```diff
@@ -0,0 +1,141 @@
+import allure
```
Comment (gigimon) [inline integration/tests/basic/test_payment_in_different_tokens.py:3]: for testing only SOL network I think we could add:
1. check that SOL balance changed for user (to verify that network uses spl token for all payments)
2. Operator got fee in Sol?
3. Check that Sol for user changed in NEON network
4. Check that user in neon network could send Sol and Neon in Sol network
5. In issue I wrote a several cases with contracts for inter-network interaction

**Review by gigimon — CHANGES_REQUESTED** [review]

File: integration/tests/basic/helpers/assert_message.py (hunk 2 above)
Comment (kristinaNikolaevaa) [reply integration/tests/basic/helpers/assert_message.py:6]: we get INSUFFICIENT_FUNDS error when there are not enough neons on account to deploy a contract, for example
and INSUFFICIENT_BALANCE error when value in transaction is bigger then account balance

**Pull request closed without merging**